### ISSUE TITLE
QLoom `>=1.3.1` doesnt remove transitive fabric dependencies

### ISSUE THREAD

**Issue by SilverAndro** [QuiltMC/quilt-loom#40]
if you pull in a dependency that uses fabric loader or fabric api, those dependencies are removed with qloom 1.2.3, but upgrading to 1.3.1 or later results in them not being removed, breaking running the game from the dev env

1.2.3:
![image](https://github.com/QuiltMC/quilt-loom/assets/52360088/43a5fbdc-46a6-4ddd-b726-d0b38b587c5d)

1.3.1 (and later):
![image](https://github.com/QuiltMC/quilt-loom/assets/52360088/e1944ed5-c7f9-4b60-97e2-7f1dde305b4c)

**Comment by TheGlitch76** [comment]
Yeah, I'm not sure what's going on here. It's the *same* patch as in 1.2.3

For now, you can exclude fabric deps manually:
```
project.configurations.all {
  exclude(group: "net.fabricmc", module: "fabric-loader")
  exclude(group: "net.fabricmc.fabric-api")
}
```

**Comment by KrLite** [comment]
> Yeah, I'm not sure what's going on here. It's the _same_ patch as in 1.2.3
> 
> For now, you can exclude fabric deps manually:
> 
> ```
> project.configurations.all {
>   exclude(group: "net.fabricmc", module: "fabric-loader")
>   exclude(group: "net.fabricmc.fabric-api")
> }
> ```

Hi! I'm experiencing the same issue with `quilt-loom 1.3.+`. The solution you provided *works,* but *only for Quilt-only projects,* since all classes from Fabric API will be unaccessible by java, causing all the code referencing Fabric API not passing the task `compileJava`.

As I'm not good on groovy at all, is there any other way to fix this problem besides downgrading the Quilt Loom version? Or, when can this issue be fixed in Quilt Loom?

**Comment by AlexIIL** [comment]
So you should be able to remove the `exclude(group: "net.fabricmc.fabric-api")` line to make it *only* exclude fabric-loader - that way you can still depend on fabric api?

**Comment by KrLite** [comment]
I've tried this but this is actually causing the... duplicate of Fabric API.

<img width="1912" alt="image" src="https://github.com/QuiltMC/quilt-loom/assets/68179735/8bafb5ca-5bcb-48f0-b3ca-8c7cc1122c50">

But  this is the closest attempt to launch.

**Comment by KrLite** [comment]
It shows that Fabric API is embedded inside the Quilted Fabric API and that's true. But in development environment, `compileJava` cannot recognize them, although no errors were emitted and you can still manually check these jars, existing.

**Comment by KrLite** [comment]
Now a new problem occurs, that is whatever the version of Quilt Loom I'm using, the Fabric API seems always unaccessible to the java compiler (I'm not excluding Fabric API neither excluding Fabric Loader). Is this another issue inside the Quilt toolchain? Recently with Quilt I've encountered many things I didn't used to understand...
